cmd/clusterctl/pkg/client/cluster/client.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"context"
+
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/internal/test"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	ctx = context.Background()
+)
+
+// Client is used to interact with a management cluster.
+// A management cluster contains following categories of objects:
+// - provider components (e.g. the CRDs, controllers, RBAC)
+// - provider inventory items (e.g. the list of installed providers/versions)
+// - provider objects (e.g. clusters, AWS clusters, machines etc.)
+type Client interface {
+	// Kubeconfig return the path to kubeconfig used to access to a management cluster.
+	Kubeconfig() string
+
+	// Kubeconfig return the K8SProxy used for operating objects in the management cluster.
+	K8SProxy() K8SProxy
```

Comment:
What do you think about renaming this to either KubeProxy or KubernetesProxy or just Proxy?

Reply:
done!